cluster-autoscaler/core/scale_down.go

```diff
@@ -163,13 +237,21 @@ func copyScaleDownResourcesLimits(source scaleDownResourcesLimits) scaleDownReso
 	return copy
 }
 
-func computeScaleDownResourcesDelta(node *apiv1.Node) (scaleDownResourcesDelta, errors.AutoscalerError) {
+func computeScaleDownResourcesDelta(node *apiv1.Node, nodeGroup cloudprovider.NodeGroup, resourcesWithLimits []string) (scaleDownResourcesDelta, errors.AutoscalerError) {
+	resultScaleDownDelta := make(scaleDownResourcesDelta)
+
 	nodeCPU, nodeMemory := getNodeCoresAndMemory(node)
+	resultScaleDownDelta[cloudprovider.ResourceNameCores] = nodeCPU
+	resultScaleDownDelta[cloudprovider.ResourceNameMemory] = nodeMemory
 
-	return scaleDownResourcesDelta{
-		cloudprovider.ResourceNameCores:  nodeCPU,
-		cloudprovider.ResourceNameMemory: nodeMemory,
-	}, nil
+	if containsGpuResources(resourcesWithLimits) {
+		gpuType, gpuCount, err := gpu.GetNodeTargetGpus(node, nodeGroup)
+		if err != nil {
+			return scaleDownResourcesDelta{}, errors.ToAutoscalerError(errors.ApiCallError, err).AddPrefix("Failed to get node %v gpu: %v", node.Name)
```

Comment:
The only place that can return an error in GetNodeTargetGpus is a cloudprovider call, so this should be a CloudProviderError.

Reply:
I changed `GetNodeTargetGpus` itself to return `AutoscalerError`. And also the two places which use the method - but after initial change the error type is not important any more.